### DISCUSSION TITLE
Excessive CPU Utilization when using Percy Playwright

### DISCUSSION THREAD

**Discussion by unlikelyzero** [percy/cli#1067] [Support]
## The problem

We are peeling the onion on some percy cli flakiness found on https://github.com/percy/cli/issues/1065 and noticed that the CPU utilization of the running process is exceeding the resources available on the basic Circle CI Agent.

CPU Utilization with `percy exec` enabled

<img width="933" alt="Screen Shot 2022-09-13 at 9 49 17 AM" src="https://user-images.githubusercontent.com/12446869/189963905-ca37b982-80f9-43e2-b41c-fd121027a03e.png">

This is the exact same test run but with `percy exec` disabled

<img width="928" alt="Screen Shot 2022-09-13 at 10 37 12 AM" src="https://user-images.githubusercontent.com/12446869/189972120-5c8b4d40-d494-4232-9e36-364aa310d60e.png">

## Environment

- Node version: 16
- `@percy/cli` version: 1.7.2
- OS version: ubuntu focal image

## Details

We are using a pretty standard percy integration with playwright. It appears that the CPU is pegged at 100% for the majority of the tests. This is after lowering playwright's cpu utilization by:
1. Disabling all diagnostic tooling from playwright (tracefile, screenshot, video, debug mode)
2. Running a single worker and single test at a time.

## Debug logs

https://app.circleci.com/pipelines/github/nasa/openmct/13351/workflows/2c27db3f-cdba-44d6-8651-0aadb149f377/jobs/50220/resources

**Comment by Robdel12** [comment]
👋🏼 Hey @unlikelyzero! I wouldn't define this as excessive given the baseline without Percy is already 50-75% or so CPU utilization. Percy absolutely does push it over the edge though. I would drop your concurrency from the default 10 to maybe 2 or 3. That will make it so there's only 2 or 3 tabs open in asset discovery processing resources. 10 tabs seems to overwhelm the resources that are available in CI. 

https://github.com/nasa/openmct/blob/master/e2e/.percy.yml
https://docs.percy.io/docs/cli-configuration#discovery

```yml

version: 2
snapshot:
  widths: [1024, 2000]
  min-height: 1440 # px
discovery:
  concurrency: 3
```

**Reply by unlikelyzero** [reply]
@Robdel12 I think this is still an issue. This is when I set discovery to 2:
<img width="831" alt="Screen Shot 2022-09-13 at 10 54 24 AM" src="https://user-images.githubusercontent.com/12446869/189975401-ef60711b-9ef7-49fb-a246-c6ee5ae128ee.png">
